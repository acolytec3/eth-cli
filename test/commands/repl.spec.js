const ReplCommand = require('../../src/commands/repl')

describe('replCommand', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'repl' with empty args and flags and throw an error.`, async () => {
    await ReplCommand.run([])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'repl --help' and throw an error.`, async () => {
    await expect(ReplCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })
})
