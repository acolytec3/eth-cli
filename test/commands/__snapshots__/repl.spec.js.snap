// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replCommand Should run 'repl' with empty args and flags and throw an error. 1`] = `
Array [
  "",
  "",
  "> ",
  "",
]
`;
